src/Workspaces/Core/Portable/Workspace/Host/PersistentStorage/DocumentKey.cs

```diff
@@ -15,7 +16,7 @@ namespace Microsoft.CodeAnalysis.Storage
     /// solution load), but querying the data is still desired.
     /// </summary>
     [DataContract]
-    internal readonly struct DocumentKey : IEqualityComparer<DocumentKey>
+    internal readonly struct DocumentKey : IEqualityComparer<DocumentKey>, IEquatable<DocumentKey>
```

Comment:
we use equality of DocumentKeys when using them as the keys in the CachedIndexMap.  So we need to actually implement equality properly so we don't get the terribly slow default struct-equality that the runtime normally does.